ads/dynad.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {validateSrcContains, validateSrcPrefix, writeScript} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function dynad(global, data) {
+  const adsrc = data.src;
```

Comment:
Can you use validateData here, it should eliminate the need for the if statement

https://github.com/ampproject/amphtml/blob/master/3p/3p.js#L225

Reply:
@calebcordry does validateData validates src attribute?

Reply:
@eduardogoncalves yes it does.